autopilot/autopilot.go

```diff
@@ -54,10 +54,10 @@ type Bus interface {
 
 	// hostdb
 	Host(ctx context.Context, hostKey types.PublicKey) (hostdb.HostInfo, error)
-	Hosts(ctx context.Context, opts api.GetHostsOptions) ([]hostdb.Host, error)
+	Hosts(ctx context.Context, opts api.HostsOptions) ([]hostdb.HostInfo, error)
```

Comment:
For me the `/search/x` endpoints were never meant to be the successor of the general restful routes you would expect to be there. I'm surprised you would deprecate `/api/bus/hosts`, that's a route that I would guess if I were to play around with the `API`.

For me the search endpoints were meant to hide away complex search logic from people that would never need it. I guess extending those options is a first step towards making the regular endpoint very similar to the complex search endpoint so maybe I'm wrong. I don't particularly like `SearchHosts` because it's not in a place where I would go and look to find information about hosts and also because it doesn't handle it defaults well (cfr `DefaultSearchHostOptions`, not sure how that happend)

I've made the change though and I marked the route for deprecation. I'm also up for reverting the change again if you change your mind about deprecating the route. I still think it makes sense to have a route that returns hosts and supports the host filter mode and pagination alongside a route that offers more complex search options.

Offtopic: maybe we should add a TODO to rename usability mode and filter mode. I don't have better alternatives now but the naming there can definitely be improved I think.

Reply:
The only reason we needed the `/search` routes really was to preserve compatibility on the early endpoints. If one endpoint is essentially a superset of another I don't see the need to keep both. So once we release v2.0 we can just make `/search/hosts` the new `/hosts` endpoint and then it will be even nicer than before. Because as you said, I wouldn't go look for info on hosts outside of `/hosts` because `/search` is really not intuitive either.